lsp-julia.el

```diff
@@ -25,33 +30,35 @@
 
 (defcustom lsp-julia-timeout 30
   "Time before lsp-mode should assume julia just ain't gonna start."
+  :type 'number
   :group 'lsp-julia)
 
 (defun lsp-julia--get-root ()
-  "Try to find the package directory by searching for a .gitignore file.
-If no .gitignore file can be found use the default directory "
-  (let ((dir (locate-dominating-file default-directory ".gitignore")))
-    (if dir
-        (expand-file-name dir)
-      default-directory)))
+  (let ((dir (locate-dominating-file default-directory "Project.toml")))
+    (if dir (expand-file-name dir)
+      (expand-file-name lsp-julia-default-environment))))
 
 (defun lsp-julia--rls-command ()
-  `(,lsp-julia-command ,@lsp-julia-flags "-e using LanguageServer; server = LanguageServer.LanguageServerInstance(STDIN, STDOUT, false); server.runlinter = true; run(server);"))
-
+  `(,lsp-julia-command
+    ,@lsp-julia-flags
+    ,(concat "-e using LanguageServer, Sockets, SymbolServer;"
+             " server = LanguageServer.LanguageServerInstance("
+             " stdin, stdout, false,"
+             " \"" (lsp-julia--get-root) "\","
+             " \"\", Dict());"
```

Comment:
Why not actually use the `DEPOT_PATH` variable? Maybe `LanguageServerInstance` doesn't do what I think it does. Is this formulation copied from https://github.com/JuliaEditorSupport/julia-vscode/blob/1ce2db32f1e639d0bd70e485837675b38c26f58d/scripts/languageserver/main.jl#L25 ?

Reply:
See here for the suggestion to use `""`: https://github.com/JuliaEditorSupport/LanguageServer.jl/issues/310

I think (not sure though) that `LanguageServer.jl` does not really support this yet:
https://github.com/JuliaEditorSupport/LanguageServer.jl/issues/316

I was thinking about this, but I was not sure about the right default value.
Also, it would have to be something like `DEPOT_PATH[1]`:
```julia
julia> server = LanguageServer.LanguageServerInstance(stdin, stdout, false, "/home/gkraemer/progs/julia/DimRedViaRegression/", DEPOT_PATH, Dict())
ERROR: MethodError: Cannot `convert` an object of type Array{String,1} to an object of type String
```